src/otx/algo/classification/metrics/hlabel_accuracy.py

```diff
@@ -72,9 +72,10 @@ def update(self, preds: torch.Tensor, target: torch.Tensor) -> None:
             target_multiclass = target[:, head_idx]
             multiclass_mask = target_multiclass > 0
 
+            # Torchmetric should be located at the same device with tensors.
+            self._metric_to_device(self.multiclass_head_accuracy[head_idx], preds.device)
```

Comment:
Removing this and can we make an overridden function for this?
https://github.com/Lightning-AI/torchmetrics/blob/7d23f974cde38ee233b71dfda006e52834604ee6/src/torchmetrics/metric.py#L776
For example,

```python
class HLabelAccuracy(Metric):
    ...
    def _apply(self, fn: Callable, exclude_state: Sequence[str] = "") -> Module:
        self.multiclass_head_accuracy = [acc._apply(fn, exclude_state) for acc in self.multiclass_head_accuracy]
        self.multilabel_accuracy = self.multilabel_accuracy._apply(fn, exclude_state)
```

Reply:
Thanks for the nice comment, done https://github.com/openvinotoolkit/training_extensions/pull/2804/commits/68abcedffe38e68b3423d28ac5f32e975c577117